Dockerfile

```diff
@@ -164,16 +166,24 @@ COPY --from=builder \
 COPY --from=builder /build/vcftools/built/bin/    /usr/local/bin/
 COPY --from=builder /build/vcftools/built/share/  /usr/local/share/
 
+ARG TARGETPLATFORM
+
+# use /env to persist variable across RUNs
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then echo "export NEXTCLADE_PLATFORM=x86_64-unknown-linux-gnu" >> /env; fi
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then echo "export NEXTCLADE_PLATFORM=aarch64-unknown-linux-gnu" >> /env; fi
+
 # Add Nextalign v2
-RUN curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextalign-x86_64-unknown-linux-gnu \
+RUN . /env; curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextalign-$NEXTCLADE_PLATFORM \
       --output /usr/local/bin/nextalign2 \
  && chmod a+rx /usr/local/bin/nextalign2
 
 # Add Nextclade v2
-RUN curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextclade-x86_64-unknown-linux-gnu \
+RUN . /env; curl -fsSL https://github.com/nextstrain/nextclade/releases/latest/download/nextclade-$NEXTCLADE_PLATFORM \
       --output /usr/local/bin/nextclade2 \
  && chmod a+rx /usr/local/bin/nextclade2
 
+RUN rm /env
```

Comment:
This results in `/env` not being present in the final image's final layer, but additional layers are still added to the final image for this `RUN` (and the two `RUN`s that create `/env`).

The number of layers matters even if the layer itself is tiny. The way to avoid inflating number of layers is to move this conditional build-only stuff into the build stage and then copy the binaries from there.

(Although maybe buildx can do fancier things here these days…?)

Reply:
In 70443f60fa6d33f9e16780db6ca5dbbd491f3a1b I got rid of the extra `/env` layers, and moved the commands up to the builder stage to access the `devel/` script.

> maybe buildx can do fancier things here these days…?

From my searching yesterday, everything I found still suggested workarounds for not having access to environment variables between `RUN` commands. 😕

Reply:
> From my searching yesterday, everything I found still suggested workarounds for not having access to environment variables between `RUN` commands. confused

Ah, I meant in terms of pruning where layer A adds file X and layer D removes file X.